test/new-e2e/tests/containers/base_test.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package containers
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
+)
+
+type baseSuite struct {
+	suite.Suite
+
+	Fakeintake *fakeintake.Client
+}
+
+func (suite *baseSuite) testMetric(metricName string, filterTags []string, expectedTags []*regexp.Regexp) {
+	suite.Run(fmt.Sprintf("%s{%s}", metricName, strings.Join(filterTags, ",")), func() {
+		suite.EventuallyWithTf(func(collect *assert.CollectT) {
+			metrics, err := suite.Fakeintake.FilterMetrics(
+				metricName,
+				fakeintake.WithTags[*aggregator.MetricSeries](filterTags),
+			)
+			if err != nil {
+				collect.Errorf("%w", err)
+				return
+			}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Looking forwards to https://github.com/stretchr/testify/pull/1481